### PR TITLE
Style search bar

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #792 

### Summary of Changes
* summary is on the right because i was not sure how to position it next to the query. Also, not sure if it would look good as you are typing
* i left the close button out because the summary is on the right and cmd+f and escape close. This feels like enough in practice
* i select the text when the search bar is opened so you have the cursor in the right spot and can delete before you type
* the results summary wraps

### CODE REVIEW

File: public/js/components/Editor.css [hunk 1]
```diff
@@ -27,23 +27,28 @@
...
 
+.search-bar ~ .editor-mount {
+  height: calc(100% - 58px)
+}
+
 .editor-mount,
 .CodeMirror {
   width: 100%;
-  height: 100%;
+  height: calc(100% - 17px);
```
Comment (jasonLaster) [inline public/js/components/Editor.css]: unfortunately, the editor was being masked by the footer. I'd like to investigate avoiding absolute positioning here. I was able to avoid it with the search bar so it *should* be possible.

**Comment by nt1m** [comment]
It would be nice if this awesome search box could make it into the built-in Editor component.

File: public/js/components/Editor.css (hunk 1 above)
Comment (jlongster) [inline public/js/components/Editor.css]: I feel like we have too many little hacks around the editor area (1px offset here and there). I would *love* it if you could take a few minutes to think about about the editor area and possibly refactor it a little bit to avoid this. Maybe a few more uses of flexbox would help. I feel like it's getting way too brittle.

I'm fine including that in this PR, it's related to the UI changes you want to make. It's hard for me to approve yet another hardcoded value :)

**Comment by jlongster** [comment]
That looks really nice! I like how it looks.

File: public/js/components/Editor.css (hunk 1 above)
Comment (jlongster) [inline public/js/components/Editor.css]: Yeah, that makes sense. Would probably be good to pull this out into a separate PR with a real fix.

I haven't looked at this CSS for a while, but seems like we could use flexbox and tell the editor-container to fill up as much as possible, while the header and footer take up as little as possible (`flex: 0`). But not sure.

**Comment by jlongster** [comment]
> It would be nice if this awesome search box could make it into the built-in Editor component.

Brian and Gabe are working on pulling out the editor into a separate module so we can all collaborate in one place and be using exactly the same thing. We might be able to pull this in there when we do it. Right now it makes sense for us to do this, because we also want to add debugger-specific options. But I agree having different experiences is not ideal... when we have separated out the editor from m-c, we can think about it.

File: public/js/components/EditorSearchBar.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  contextTypes: {
+    shortcuts: PropTypes.object
+  },
+
+  setupKeyboardShortcuts() {
+    if (this.keyShortcutsEnabled) {
+      return;
+    }
+
+    this.keyShortcutsEnabled = true;
```
Comment (jlongster) [inline public/js/components/EditorSearchBar.js]: Why keep track of this? Wouldn't just calling this in `componentDidMount` work instead of `componentDidUpdate`?

File: public/js/components/EditorSearchBar.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  render() {
+    if (!isEnabled("search") || !this.state.enabled) {
+      return dom.div();
+    }
+
+    const { count } = this.state;
+
+    return dom.div(
+      { className: "search-bar" },
+      new Svg("magnifying-glass"),
```
Comment (jlongster) [inline public/js/components/EditorSearchBar.js]: You don't need `new` here

File: public/js/components/EditorSearchBar.js (hunk 2 above)
Comment (jlongster) [inline public/js/components/EditorSearchBar.js]: "needed the window"? Shouldn't `window` always exist? I encourage you to go down these rabbit holes :) Otherwise we are just accumulating technical debt.

**Comment by jlongster** [comment]
Awesome, thanks!

File: public/js/components/EditorSearchBar.js (hunk 2 above)
Comment (jasonLaster) [inline public/js/components/EditorSearchBar.js]: Yep - all good points. I can absolutely see where you're coming from.
